src/app/search/search.component.ts

```diff
@@ -0,0 +1,142 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { CmsService, HalDoc } from '../core';
+import { StoryModel, SeriesModel } from '../shared';
+
+@Component({
+  selector: 'publish-search',
+  styleUrls: ['search.component.css'],
+  templateUrl: 'search.component.html'
+})
+
+export class SearchComponent implements OnInit {
+
+  selectedTab: string;
+
+  isLoaded = false;
+  totalCount: number;
+  pages: number[];
+  noStories: boolean;
+  auth: HalDoc;
+  stories: StoryModel[];
+  storyLoaders: boolean[];
+
+  currentPage: number;
+  showNumPages: number = 10;
+  pagesBegin: number;
+  pagesEnd: number;
+
+  searchText: string;
+  searchSeriesId: number;
+  searchSeries: SeriesModel;
+  allSeriesIds: number[];
+  allSeries: any;
+
+  searchOrderBy: string;
+  orderByOptions: any[] = [
+    {
+      id: 'TITLE',
+      name: 'Story Title'
+    },
+    {
+      id: 'UPDATED',
+      name: 'Last Updated'
+    },
+    {
+      id: 'PUBLISHED',
+      name: 'When Published'
+    }
+  ];
+
+  constructor(private cms: CmsService,
+              private route: ActivatedRoute) {}
+
+  ngOnInit() {
+    this.route.params.subscribe(params => {
+      this.selectedTab = params['tab'];
+      this.searchSeriesId = +params['seriesId'];
+
+      this.cms.follow('prx:authorization').subscribe((auth) => {
+        this.auth = auth;
+
+        this.allSeriesIds = [-1];
+        this.allSeries = {};
+        auth.followItems('prx:series', {filters: 'v4'}).subscribe((series) => {
+          for (let s of <HalDoc[]> series) {
+            this.allSeriesIds.push(s.id);
+            this.allSeries[s.id] = new SeriesModel(auth, s, false);
+          }
+
+          this.searchText = '';
+          this.searchSeries = null;
+          this.currentPage = 1;
+          if (this.searchSeriesId) {
+            this.searchBySeries();
+          } else {
+            this.search(1);
+          }
+        });
+      });
+    });
+  }
+
+  searchBySeries() {
+    this.searchSeries = this.allSeries[this.searchSeriesId];
+    this.search(1);
+  }
+
+  search(page: number, per = 12) {
+    this.currentPage = page;
+    this.isLoaded = false;
+    this.noStories = false;
+
+    let parent = this.searchSeries ? this.searchSeries.doc : this.auth;
+
+    let filters = ['v4'];
+    if (+this.searchSeriesId === -1) {
+      filters.push('noseries');
+    }
+    if (this.searchText) {
+      filters.push('text=' + this.searchText);
+    }
+    let params = {page: this.currentPage, per, filters: filters.join(',')};
+
+    let storiesCount = parent.count('prx:stories') || 0; // wrong, doesn't account for filter. looks obvs wrong with No Series filter
+    if (storiesCount > 0) {
+      this.storyLoaders = Array(storiesCount);
```

Comment:
Looks like `storyLoaders` never gets reset, if the current search is returning 0 stories.  (So you get > 1 loading-spinners as you run the search).